### PR TITLE
Add UI for selecting multiple wards from /reports/ body page

### PR DESCRIPTION
**Pull request by davea**
multi_wards

Fixes mysociety/fixmystreetforcouncils#231

### CODE REVIEW

File: web/cobrands/fixmystreet/fixmystreet.js [hunk 1]
```diff
@@ -1118,6 +1126,7 @@ fixmystreet.display = {
                 $sideReport.find('#key-tool-problems-nearby').addClass('js-back-to-report-list');
             }
             fixmystreet.set_up.map_sidebar_key_tools();
+            fixmystreet.set_up.ward_select_multiple();
```
Comment (dracos) [inline web/cobrands/fixmystreet/fixmystreet.js]: This won't do anything, you're switching to a individual report page that lacks this link.
The main page is then hidden, so you don't need to reinstate it when you go back either, it looks like.

File: templates/web/base/reports/_ward-list.html [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
+          </label>
+          <a class="js-ward-single" href="[% child.url %]">[% child.name %]</a>
         </li>
       [% END %]
     </ul>
     </section>
+    <p>
+      <input class="btn js-ward-multi hidden" type="submit" value="[% loc('View wards') %]" />
+    </p>
+    <p class="js-ward-single"><a href="#" class="js-ward-select-multiple">[% loc('Select multiple wards?') %]</a></p>
```
Comment (dracos) [inline templates/web/base/reports/_ward-list.html]: Can we have this at the top? Bit hard to find at the bottom.

Also the button shouldn't be viewable unless JS is enabled – or actually *ideally*, the button would submit to the server and reshow the page with the checkboxes shown, please :)

File: perllib/FixMyStreet/App/Controller/Reports.pm [hunk 3]
```diff
@@ -126,6 +126,17 @@ sub ward : Path : Args(2) {
     my @wards = split /\|/, $ward || "";
     $c->forward( 'body_check', [ $body ] );
 
+    # If viewing multiple wards, rewrite the url from
+    # /reports/Borsetshire?ward=North&ward=East
+    # to
+    # /reports/Borsetshire/North|East
+    my @ward_params = $c->get_param_list('ward');
+    if ( @ward_params ) {
+        $c->stash->{ward_names} = [ @wards, @ward_params ];
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Reports.pm]: Any reason you didn't use `wards` here - it looks like you do want to pass them through short_name for escaping purposes, given it's the names coming through?

File: templates/web/base/reports/_ward-list.html [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+    <p class="js-ward-single">[% loc('Follow a ward link to view only reports within that ward.') %]</p>
+    <p class="js-ward-multi hidden">[% loc('Select multiple wards to view only reports within those wards.') %]</p>
+    <form method="get" action="">
     <section class="full-width">
     <ul class="item-list item-list--wards">
       [% FOR child IN children.values.sort('name') %]
         <li class="item-list__item item-list--wards__item">
-            <a href="[% child.url %]">[% child.name %]</a>
+          <label class="js-ward-multi hidden">
+            <input type="checkbox" name="ward" value="[% child.name | html %]">
```
Comment (dracos) [inline templates/web/base/reports/_ward-list.html]: Alternatively to my previous comment, if you stored the short name (last bit of the URL) here you could just pass them through.

File: templates/web/base/reports/_ward-list.html (hunk 2 above)
Comment (dracos) [reply templates/web/base/reports/_ward-list.html]: How about f26feef26d8a1961b124869d47aa4c749f25c5af? :)

Comment (davea) [reply templates/web/base/reports/_ward-list.html]: That'll do nicely!

**Pull request merged**